### DISCUSSION TITLE
virtual library built only for executables?

### DISCUSSION THREAD

**Discussion by progman1** [ocaml/dune#15151] [Q&A]
that's what the docs talk about.
as the entity dune is building for me is meant to be just a library, not an exec,
can I do that with a virtual component?
I'm trying to build the same target with different files.

**Comment by progman1** [comment]
I'm proposing the use of vlibs in targets other than executables.
it seems a concrete library that specifies an implementation for a vlib is a legitimate target, no?
but as I haven't gotten it to work I assume, in the first place, that the docs are complete and that such a target is not supported, and if that assumption is wrong then this is a feature request.

**Comment by ghost** [comment]
Note that you can depend on a concrete implementation in a library
